apps/frontend/core-dapp/src/features/simulator/SimulatorSettings.tsx

```diff
@@ -0,0 +1,156 @@
+import { Input as BaseInput, spacingIncrement } from 'prepo-ui'
+import { InputContainer, LabelContainer } from 'prepo-ui/src/components/Input'
+import styled from 'styled-components'
+import { RangeValidity } from './SimulatorPage'
+import Card from '../../components/Card'
+import { FormValue } from '../../hooks/useFormValue'
+
+const ALLOWED_DECIMAL_SEPARATORS = ['.', ',']
+
+const Wrapper = styled(Card)`
+  .ant-card-body {
+    display: flex;
+    flex-direction: column;
+    gap: ${spacingIncrement(18)};
+    :after,
+    :before {
+      display: none;
+    }
+  }
+  width: 100%;
+`
+
+const InputGroup = styled.fieldset``
+
+const Label = styled.legend`
+  color: ${({ theme }): string => theme.color.neutral2};
+  font-size: ${({ theme }): string => theme.fontSize.xs};
+  margin-bottom: ${spacingIncrement(4)};
+`
+
+const Inputs = styled.div`
+  display: flex;
+  gap: ${spacingIncrement(8)};
+`
+
+const Input = styled(BaseInput).attrs({ size: 'small' })`
+  ${InputContainer} {
+    border-radius: ${({ theme }): string => theme.borderRadius.xs};
+    padding: ${spacingIncrement(8)} ${spacingIncrement(12)};
+  }
+
+  ${LabelContainer} {
+    display: none;
+  }
+
+  && {
+    .ant-input {
+      font-size: ${({ theme }): string => theme.fontSize.sm};
+      font-weight: ${({ theme }): number => theme.fontWeight.medium};
+      line-height: ${spacingIncrement(20)};
+    }
+
+    .ant-input-suffix {
+      color: ${({ theme }) => theme.color.neutral1};
+    }
+  }
+`
+
+export const Title = styled.h2`
+  color: ${({ theme }): string => theme.color.secondary};
+  font-size: ${({ theme }): string => theme.fontSize.base};
+  font-weight: ${({ theme }): number => theme.fontWeight.semiBold};
+`
+
+const NumberInput: React.FC<{
+  isValid: boolean
+  max: number
+  min: number
+  suffix: string
+  value: FormValue<string>
+}> = ({ isValid, max, min, suffix, value: { value, setValue } }) => (
+  <Input
+    error={!isValid}
```

Comment:
Should add block to keep it fill the width

```suggestion
  <Input
    block
    error={!isValid}
```